### PR TITLE
Cs/sc 2186 add direct link to superset from hq

### PR DESCRIPTION
**Pull request by Charl1996**
## Product Description
This PR adds another item (subject to the FF being enabled) to the Data -> sidebar, called "CommCare Analytics", as well as to the Data -> dropdown menu. When a user clicks on either of these items, the user is navigated to a page which looks like the following:
image

When a user click's the button the user is navigated to Superset in a new browser tab.

## Technical Summary
[Ticket](https://dimagi-dev.atlassian.net/browse/SC-2186)

The redirect URLs are different for production and staging.
Production -> http://commcare-analytics.dimagi.com
Staging -> http://commcare-analytics-demo.dimagi.com

Beside the aforementioned, I also attempted a little bit of refactoring of the `sidebar_items` in corehq/tabs/tabclasses.py.

## Feature Flag
[Activates Analytics features to create Superset based reports and dashboards using UCR data](https://staging.commcarehq.org/hq/flags/edit/superset-analytics/)

### Safety story
Tested locally

### Automated test coverage
No automated tests necessary I think

### QA Plan
Don't think a formal QA plan is necessary?

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by Charl1996** [comment]
I have noticed that when I disable the FF, the dropdown menu still shows the link to the CommCare Analytics page, but disappears after a while (not sure why). I'm now looking into how to "refresh" the dropdown menu on the fly.

File: corehq/apps/export/views/list.py [hunk 1]
```diff
@@ -1050,3 +1051,22 @@ def page_context(self):
...
+    @property
+    def page_context(self):
+        return {
+            "analytics_redirect_url": self.commcare_analytics_url
+        }
+
+    @property
+    def commcare_analytics_url(self):
+        url = "https://commcare-analytics.dimagi.com" \
+            if settings.SERVER_ENVIRONMENT == "production" else "http://commcare-analytics-demo.dimagi.com/"
```
Comment (sravfeyn) [inline corehq/apps/export/views/list.py:1071]: This could be a new setting altogether such as `settings.SUPERSET_HOST`

**Comment by sravfeyn** [comment]
> I have noticed that when I disable the FF, the dropdown menu still shows the link to the CommCare Analytics page, but disappears after a while (not sure why). I'm now looking into how to "refresh" the dropdown menu on the fly.

I have noticed this behaviour with other feature flags as well. It's likely a cache issue.

**Comment by Charl1996** [comment]
> This is something that affects almost all pages of the site, so testing is crucial, unless you are clear of the impact of the refactor.

Yeah, the failing tests indicates that I did not test this thoroughly enough. Currently busy going through it again.

**Comment by Charl1996** [comment]
Maybe I should separate the refactor into a different branch. That way it can stay on staging for longer for the "passive QA" whilst not impeding the progress of the feature in the ticket.

**Comment by Charl1996** [comment]
Closign in favour of [this](https://github.com/dimagi/commcare-hq/pull/32138) PR. (basically split out the refactor)

**Pull request closed without merging**